File: LAB/P1/integrals-java/Function.java

class Function implements Expresion{
  private Operation operation;
  private Expresion leftExpresion;
  private Expresion rightExpresion;
  public Function (Operation operation, Expresion leftExpresion, Expresion rightExpresion) {
    this.operation = operation;
    this.leftExpresion = leftExpresion;
    this.rightExpresion = rightExpresion;
  }

  public double evaluate (double value) {
    return operation.evaluate(leftExpresion.evaluate(value), rightExpresion.evaluate(value));
  }
}

File: LAB/P1/integrals-js/Function.js

class Function {

  constructor (operation, leftFunction, rightFunction) {
    this.fun = operation
    this.leftFunction = leftFunction
    this.rightFunction = rightFunction 
  }

  evaluate (value) {
    return this.fun.evaluate(this.leftFunction.evaluate(value), this.rightFunction.evaluate(value))
  }
}

module.exports = Function